[Learn] Added bookmark to learning paths

NikhilSharma03 opened the pull request:
Signed-off-by: Nikhil Sharma <nikhilsharmamusic2000@gmail.com>

**Description**

- Auto bookmarking in learning paths

https://user-images.githubusercontent.com/58226527/151178727-51ccf0d8-7913-4bc7-b824-3a037cc6a5ea.mp4

**Notes for Reviewers**


**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 

<!--
Thank you for contributing to Layer5 projects! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR. 
3. Sign your commits

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

l5io commented:
🚀 Preview for commit 9457b5c2356f4baf57b94b148adc09b5be3b90b1 at: https://61f159772d8d55b8f867de9b--layer5ng.netlify.app

Nikhil-Ladha commented on src/sections/Learn-Layer5/Course-Overview/courseoverview.style.js:
```diff
@@ -107,12 +107,40 @@ export const CourseOverviewWrapper = styled.div`
+  .resume-button-container{
+    display: inline-block;
+    margin-left: 1rem;
+    animation: bounce .5s 2;
```
Instead of animating it, change the background color to `${props => props.theme.highlightColor}` which could also be done by just adding `secondary` prop to the `Button` component like:
`<Button secondary utl="...." title="....." />`

NikhilSharma03 replied:
Got it.

Nikhil-Ladha commented on src/sections/Learn-Layer5/Course-Overview/index.js:
```diff
@@ -81,6 +91,13 @@ const CourseOverview = ({ course, chapters, serviceMeshesList }) => {
             title="Get Started"
             url={`istio/${course.frontmatter.toc[0]}`}
           />
```
Another good thing will be, to update the title of this button to `Start again` or something similar (@leecalcote might help with this) if there is already a bookmark (a check similar to the `resume` button would do`).

Nikhil-Ladha replied:
Keep this one as `Resume` only.
Make the above button `Start Again` when there is a bookmark.
So, there will be 2 cases:
1. When  a new user comes he/she should see only `Get Started` button
2. When a returning user comes, he/she should see `Start again` and `Resume` button.

Nikhil-Ladha replied:
Adjust the code to accommodate both of the above cases.

NikhilSharma03 replied:
Oh sorry. Got it. I will make the necessary changes.

NikhilSharma03 commented on src/sections/Learn-Layer5/Course-Overview/index.js:
```diff
@@ -81,6 +91,13 @@ const CourseOverview = ({ course, chapters, serviceMeshesList }) => {
             url={`istio/${course.frontmatter.toc[0]}`}
           />
+        {bookmarkUrl.split("/")[4] === course.fields.slug.split("/")[3] && hasBookmark && (
+          <div className="resume-button-container"><Button
```
To align the `Resume` button the inline property it stands next to the `Get Started` button. I will add the styles directly to Buttons.

NikhilSharma03 commented on src/sections/Learn-Layer5/Course-Overview/index.js:
```diff
@@ -40,17 +42,25 @@ const CourseOverview = ({ course, chapters, serviceMeshesList }) => {
-            />
-          </div>
-        </>
+        <Link to={sm+"/getting-started"}>
+           <div className="service-mesh-courses" key={index}>
+             <Image
+               {...findServiceMeshImage(serviceMeshImages, sm).imagepath}
+               className="docker"
+               alt={sm}
+             />
+           </div>
+        </Link>
```
Ok. @leecalcote once that we can make these buttons clickable.

NikhilSharma03 commented:
#### Latest Changes

- Removed the extra div
- Removed the animation and added `secondary` color to the button (Start Again)
- Reverted back the Button/link icons
- Changed `Resume` title to `Start Again`

@Nikhil-Ladha Can you please review?

l5io commented:
🚀 Preview for commit 2ac8594b9b58f43b02456ec1abb18785eec5cd5a at: https://61f29526d8a2010058718d81--layer5ng.netlify.app

NikhilSharma03 commented:
@Nikhil-Ladha Done.

l5io commented:
🚀 Preview for commit db107cbbcb0fc5ca22e0ce1797ee7781ea843487 at: https://61f2a926a7ea530acadec316--layer5ng.netlify.app

Nikhil-Ladha commented on src/sections/Learn-Layer5/Course-Overview/index.js:
```diff
@@ -78,9 +86,17 @@ const CourseOverview = ({ course, chapters, serviceMeshesList }) => {
+        {hasBookmark && (
+          <Button
+            className="start-again-button"
+            secondary
```
Sorry for the trouble, but it should be `primary`.
Also, this color will get overridden by the CSS at line: 34, so please fix that as well
Such that the final state should be:
1. For newcomer: `Get Started` <- White Color
2. For returning user: `Start again` <- White Color, `Resume` <- Saffron Color

NikhilSharma03 replied:
Done
